pkg/datastore/mysql/iterator.go

```diff
@@ -35,10 +40,89 @@ func (it *Iterator) Next(dst interface{}) error {
 	if err != nil {
 		return err
 	}
+
+	// Update last iterated item as last read row.
+	it.last = &rowDataConverter{val: val}
+
 	return decodeJSONValue(val, dst)
 }
 
-// Cursor implementation for MySQL Iterator
+// Cursor builds a base 64 string (encode from string in map[string]interface{} format).
+// The cursor contains only values attached with the fields used
+// as ordering fields.
 func (it *Iterator) Cursor() (string, error) {
-	return "", datastore.ErrUnimplemented
+	if it.last == nil {
+		return "", datastore.ErrInvalidCursor
+	}
+
+	lastObjData := it.last.Data()
+
+	cursor := make(map[string]interface{})
+	for _, o := range it.orders {
+		val, ok := lastObjData[o.Field]
+		if !ok {
+			return "", datastore.ErrInvalidCursor
+		}
+		// TODO: Support build cursor from nested Ordering field.
+		cursor[o.Field] = val
+	}
+
+	b, _ := json.Marshal(cursor)
+	return base64.StdEncoding.EncodeToString(b), nil
+}
+
+type DataConverter interface {
```

Comment:
I would think this isn't needed to be public. Firestore's one as well.

Reply:
Sure 👍 I was thinking to move this interface under `datastore` package and it will be shared to both this `mysql` and `firestore` package 👀 How do you think about that 🤔 ( Will address it by a separated PR )

Reply:
Honestly, I'd say it doesn't need to be an interface (just be struct is still ok) anymore.

Reply:
It's because that is generated and used within only this package. Looks like a struct is enough.

Reply:
I think it's worth it, making it as an interface is for testing 👀 
ref: pkg/datastore/mysql/iterator_test.go or [firestore/iterator_test.go](https://github.com/pipe-cd/pipe/blob/master/pkg/datastore/firestore/iterator_test.go#L26) - for firestore iterator 👀

Reply:
I hear you. makes sense. Okay I got the value of making it an interface 👍 
Still I'm wondering if it should be a common one or not. Let's talk about it on another PR.

Reply:
Nice question 👍 Let's discuss on merged interface PR 😁